### PR TITLE
Fix missing names in ExternalMDSnapshot

### PR DESCRIPTION
**Pull request by dwhswenson**
`ExternalMDSnapshot` was moved to a different file, and didn't get all its module-level names redefined. Bug uncovered by @jasperdenduijf in #907.

~~Right now, this still requires tests~~ (tests added!). We're missing coverage on the lines that caused that part of the issue in #907, so we should have tests covering those lines.

### CODE REVIEW

**Comment by sefalkner** [comment]
Tested this on my Gromacs TPS setup that also had these occasional Errors (#907) after some 10-30 mc cycles, now runs smooth for > 300 cycles.

In general I thought about contributing to the docs for the gromacs engine. For a new user it is unclear that setting n_frames_max to the gromacs nsteps option is crucial as there is little documentation about it.

**Comment by dwhswenson** [comment]
> Tested this on my Gromacs TPS setup that also had these occasional Errors (#907) after some 10-30 mc cycles, now runs smooth for > 300 cycles.

Thanks for the confirmation that this fixes -- I'll make finishing this PR a priority, but things are a little bit busy for the next week. (Out-of-town guests.)

> In general I thought about contributing to the docs for the gromacs engine. For a new user it is unclear that setting n_frames_max to the gromacs nsteps option is crucial as there is little documentation about it.

Please do! Improvements to the docs are always welcome. External engines (such as Gromacs) are newer, and neither the code nor the docs have had as much stress testing, so the help is appreciated.

That said, you might wait until, say, Monday. I've been working on a branch to improve a few things in the Gromacs engine docs -- docstrings and, especially, the example notebooks -- so it might be easier to build on that than to write (possibly conflicting) changes. I'll try to make a PR of that very soon.

**Comment by jasperdenduijf** [comment]
This might be an obvious question, but once I've installed openpathsampling, how do I pull these commits and add them to my current PATH?

**Comment by dwhswenson** [comment]
@jasperdenduijf : I think most of what you want is described at http://openpathsampling.org/latest/developers/install_developers.html (after getting the dev install script with the curl command at http://openpathsampling.org/latest/install.html#developer-install-with-conda). The quickest way to do a bugfix test install is under the heading "Quick bugfix/developer installation" in the first link. That creates a separate conda environment for the bugfix version of OPS. For this specific branch, you'd use remote `dwhswenson` and branch `fix_ext_snapshot_missing` (in place of `experimental_feature` in the docs).

Assuming the tests that I just added pass, then I expect this will be merged to master tomorrow evening or Saturday morning. It will be included in the 1.3 release, which I hope to release early next week.

**Comment by dwhswenson** [comment]
Tests pass; I'll leave this open for comment for at least 24 hours before merging. I will merge no earlier than Friday 20 March, 14:00 GMT (15:00 my local).

This PR involves adding some missing imports and then adding tests to improve our test coverage, so I can't imagine anything here is controversial.

File: openpathsampling/tests/test_external_snapshots.py [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+            snap.load_details()
+            expected_xyz = np.array([[0.1 * i, 0.0]])
+            expected_vel = np.array([[1.0, 0.0]])
+            np.testing.assert_array_equal(snap._xyz, expected_xyz)
+            np.testing.assert_array_equal(snap._velocities, expected_vel)
+            np.testing.assert_array_equal(snap._box_vectors, self.box)
+
+    def test_load_details_indexerror(self):
+        engine = ErrorMockEngine(self.engine.sequences)
+        engine.accessed
```
Comment (sroet) [inline openpathsampling/tests/test_external_snapshots.py]: why is this line here? 
just to test that the attribute is there? 
or is it leftover debug code?

Comment (dwhswenson) [reply openpathsampling/tests/test_external_snapshots.py]: Thanks  for catching this: It was there because `ErrorMockEngine` had an `__init_` method instead of an `__init__` method, and it took me an embarrassingly long time to track down the missing underscore. This was part of that process. Removed!

Comment (sroet) [reply openpathsampling/tests/test_external_snapshots.py]: ah, classic :laughing:
